app/grandchallenge/documentation/models.py

```diff
@@ -0,0 +1,136 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import models
+from django.db.models import Max
+from django.utils.translation import gettext_lazy as _
+from django_extensions.db.fields import AutoSlugField
+from simple_history.models import HistoricalRecords
+
+from grandchallenge.core.templatetags.bleach import clean
+from grandchallenge.core.utils.query import index
+from grandchallenge.subdomains.utils import reverse
+
+
+class LevelChoices(models.TextChoices):
+    """Documentation page level options"""
+
+    PRIMARY = "PRIMARY", _("Top level")
+    SECONDARY = "SECONDARY", _("Second level")
+    TERTIARY = "TERTIARY", _("Third level")
+
+
+class DocPageLevel:
+    """Documentation page level."""
+
+    DocPageLevelChoices = LevelChoices
+
+
+class DocPage(models.Model):
+
+    Level = DocPageLevel.DocPageLevelChoices
+
+    UP = "UP"
+    DOWN = "DOWN"
+    FIRST = "FIRST"
+    LAST = "LAST"
+
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    title = models.CharField(max_length=1024)
+    slug = AutoSlugField(populate_from="title", max_length=1024)
+    display_title = models.CharField(
+        max_length=255,
+        default="",
+        blank=True,
+        help_text=(
+            "On pages and in menu items, use this text. Spaces and special "
+            "chars allowed here. Optional field. If emtpy, title is used"
+        ),
+    )
+
+    content = models.TextField()
+
+    level = models.CharField(
+        max_length=20,
+        choices=Level.choices,
+        default=Level.PRIMARY,
+        help_text=(
+            "As which level should this page be displayed in the sidebar?"
+        ),
+    )
+
+    order = models.IntegerField(
+        editable=False,
+        default=1,
+        help_text="Determines order in which pages appear in side menu",
+    )
+
+    history = HistoricalRecords()

```

Comment:
You probably only want to keep a few fields in the history. For instance, only the content.